contracts/src/v0.8/automation/chains/OptimismModuleV2.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.19;
+
+import {GasPriceOracle as OVM_GasPriceOracle} from "../../vendor/@eth-optimism/contracts-bedrock/v0.17.3/src/L2/GasPriceOracle.sol";
+import {ChainModuleBase} from "./ChainModuleBase.sol";
+import {ConfirmedOwner} from "../../shared/access/ConfirmedOwner.sol";
+
+/**
+ * @notice OptimismModuleV2 provides a cost-efficient way to get L1 fee on OP stack.
+ * After EIP-4844 is implemented in OP stack, the new OP upgrade includes a new function getL1FeeUpperBound to estimate
+ * the upper bound of current transaction's L1 fee.
+ */
+contract OptimismModuleV2 is ChainModuleBase, ConfirmedOwner {
+  error InvalidL1FeeCoefficient(uint8 coefficient);
+  event L1FeeCoefficientSet(uint8 coefficient);
+
+  /// @dev OVM_GASPRICEORACLE_ADDR is the address of the OVM_GasPriceOracle precompile on Optimism.
+  /// @dev reference: https://community.optimism.io/docs/developers/build/transaction-fees/#estimating-the-l1-data-fee
+  address private constant OVM_GASPRICEORACLE_ADDR = address(0x420000000000000000000000000000000000000F);
+  OVM_GasPriceOracle private constant OVM_GASPRICEORACLE = OVM_GasPriceOracle(OVM_GASPRICEORACLE_ADDR);
+
+  /// @dev L1 fee coefficient can be applied to reduce possibly inflated gas price
```

Comment:
What caused the "inflated" price? Are we trying to account for the data compression by using this coefficient?

Reply:
yes. that is exactly what it means. maybe this can be better documented. will update